cmd/kube-scheduler/app/options/options.go

```diff
@@ -150,13 +150,26 @@ func newDefaultComponentConfig() (*kubeschedulerconfig.KubeSchedulerConfiguratio
 // Flags returns flags for a specific scheduler by section name
 func (o *Options) Flags() (nfs cliflag.NamedFlagSets) {
 	fs := nfs.FlagSet("misc")
-	fs.StringVar(&o.ConfigFile, "config", o.ConfigFile, `The path to the configuration file. The following flags can overwrite fields in this file:
+	fs.StringVar(&o.ConfigFile, "config", o.ConfigFile, `The path to the configuration file. The following flags will be not effective when set --config , we must move into this file:
```

Comment:
I think @alculquicondor's intent was to keep the original (shorter) list of flags that *do* work with `--config`, and only note in the description of those that don't work.

This change doesn't provide any info to the user about what they *can* set with the config flag, and having the list of ones that don't work is redundant with having that info added to the flag description, as you have done

Reply:
+1

Reply:
ok, your suggestion is really pertinent. We need to care about the parameters that are in effect, and remind users to pay attention to those that are not.  : )

Reply:
1. The command line parameters that are still effective when --config is set, need to remove --port and --address。
2.

Reply:
@changshuchao I see, then in my opinion it makes sense to note here which flags will not be effective for the flags which we import from other packages (and therefore can't update the description of).

Are all of the flags you listed in your comment imported? some like `scheduler-name` we should have the ability to update the description right? Though, it may be less confusing to have all of the affected flags in one spot. In other words, as a user I would expect this list to be complete if it's here at all.

So, since we don't have control over every flag's description that isn't effective with the config file I think it does make sense to list them all here. Sorry to flip flop on that, I wasn't originally thinking about the ones we can't update the description for (@alculquicondor may have missed that too, do you agree Aldo?)

Reply:
@damemi @alculquicondor

Reply:
@damemi  
'Are all of the flags you listed in your comment imported?' --Yes,i think it's all included.
'some like scheduler-name we should have the ability to update the description right? ' --Yes, I have modified everything that can be modified.